HPXMLtoOpenStudio/resources/battery.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+class Battery
+  def self.apply(model, battery)
+    obj_name = battery.id
+
+    power = battery.rated_power_output # kW
+    capacity = battery.nominal_capacity # kWh
+    voltage = battery.nominal_voltage # V
+
+    return if power <= 0 || capacity <= 0 || voltage <= 0
+
+    is_outside = (battery.location == HPXML::LocationOutside)
+    if not is_outside
+      frac_sens = 1.0
+    else # Internal gains outside unit
+      frac_sens = 0.0
+    end
+
+    # The following calculations are from Rohit C.
+    number_of_cells_in_series = Integer((voltage / 3.6).round)
+    number_of_strings_in_parallel = Integer(((capacity * 1000.0) / (voltage * 3.2)).round)
+    battery_mass = (capacity / 10.0) * 99.0 # kg
+    battery_surface_area = 0.306 * (capacity**(2.0 / 3.0)) # m^2
+    c_rate = power / capacity
+
+    minimum_storage_state_of_charge_fraction = 0.15 # from SAM
+    maximum_storage_state_of_charge_fraction = 0.95 # from SAM
+    initial_fractional_state_of_charge = minimum_storage_state_of_charge_fraction
+
+    elcds = model.getElectricLoadCenterDistributions
+    if elcds.empty?
+      elcd = OpenStudio::Model::ElectricLoadCenterDistribution.new(model)
+      elcd.setName('Battery elec load center dist')
+      elcd.setElectricalBussType('AlternatingCurrentWithStorage')
+    else # has PV
+      elcd = elcds[0]
+
+      if elcd.inverter.is_initialized
+        elcd.setElectricalBussType('DirectCurrentWithInverterDCStorage')
+        initial_fractional_state_of_charge = 0.5 # from SAM
+      end
+    end
```

Comment:
This code needs to be updated because [there can also be an ELCD if there is a generator](https://github.com/NREL/OpenStudio-HPXML/blob/b00e22651658644db2e7dd7160082dadbf7f0e48/HPXMLtoOpenStudio/resources/generator.rb#L44-L48) (e.g., a propane generator).

Reply:
I suppose that brings up the same issue as we had with PV: what to due when there are two generators and thus two ELCD objects?

Reply:
Maybe another scenario could be when there is generator + pv + battery. Which ELCD does battery go on?

Reply:
If there is a PV system, always attach the battery to its ELCD. If there's no PV system, either use a standalone ELCD or don't model the battery at all (they should give the same result). In other words, ignore the generator ELCD.

Reply:
I think if there's a battery and no PV, we just throw a warning and leave the battery out for now.